lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -532,15 +550,109 @@ defmodule ExUnit.Case do
       end
 
   """
-  def register_attribute(env, name, opts \\ [])
+  @spec register_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_attribute(env, name, opts \\ []) do
+    do_register_attribute(:ex_unit_registered_test_attributes, env, name, opts)
+  end
+
+  @doc """
+  Registers a new describe attribute to be used during `ExUnit.Case` tests.
+
+  The attribute values will be available through `context.registered`.
+  Registered values are cleared after each `ExUnit.Case.describe/2` similar
+  to `@describetag`.
+
+  `Module.register_attribute/3` is used to register the attribute,
+  this function takes the same options.
+
+  ## Examples
+
+      defmodule MyTest do
+        use ExUnit.Case
+
+        ExUnit.Case.register_describe_attribute(__MODULE__, :describe_fixtures, accumulate: true)
+
+        describe "using custom attribute" do
+          @describe_fixtures :user
+          @describe_fixtures {:post, insert: false}
+
+          test "has attribute", context do
+            assert context.registered.describe_fixtures == [{:post, insert: false}, :user]
+          end
+        end
+
+        describe "custom attributes are cleared per describe" do
+          test "doesn't have attributes", context do
+            assert context.registered.describe_fixtures == []
+          end
+        end
+      end
+
+  """
+  @spec register_describe_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_describe_attribute(env, name, opts \\ []) do
+    do_register_attribute(:ex_unit_registered_describe_attributes, env, name, opts)
```

Comment:
We moved away from `do_` prefixes in Elixir for private functions. Since the arity of the functions is different, you can call the private function `register_attribute/4`.